src/local_sensitivity/interpolating_adjoint.jl

```diff
@@ -27,17 +29,31 @@ function ODEInterpolatingAdjointSensitivityFunction(g,sensealg,discrete,sol,dg,c
     tspan[1] > interval_end && push!(intervals, (interval_end, tspan[1]))
     cursor = lastindex(intervals)
     interval = intervals[cursor]
-    cpsol = solve(remake(sol.prob, tspan=interval, u0=sol(interval[1])), sol.alg; tols...)
+
+    if typeof(sol.prob) <: SDEProblem
+      # replicated noise
+      _sol = deepcopy(sol)
+      idx1 = searchsortedfirst(_sol.t, interval[1]-1000eps(interval[1]))
+      idx2 = searchsortedfirst(_sol.t, interval[2])
+      forwardnoise = DiffEqNoiseProcess.NoiseGrid(_sol.t[idx1:idx2], _sol.W.W[idx1:idx2])
+      dt = abs(_sol.W.dt)
+      if dt < 10000eps(_sol.t[end])
```

Comment:
10000 is a bit high. Is 100 not sufficient?

Reply:
100 doesn't change dt in the test case. 1000 seems to work..  It is not yet optimal.. the unique check is also sensitive to the discretization.. for 1e-9 for example, the current version would need a different number of digits when checking the uniqueness ..

```Julia
using DiffEqSensitivity, StochasticDiffEq
using Random
seed = 100
Random.seed!(seed)
u₀ = [0.5]
tstart = 0.0
tend = 0.1
dt = 0.005
trange = (tstart, tend)
t = tstart:dt:tend
tarray = collect(t)

function g(u,p,t)
  sum(u.^2.0/2.0)
end

function dg!(out,u,p,t,i)
  (out.=-u)
end

p2 = [1.01,0.87]


f_oop_linear(u,p,t) = p[1]*u
σ_oop_linear(u,p,t) = p[2]*u

dt1 = tend/1e4

Random.seed!(seed)
prob_oop = SDEProblem(f_oop_linear,σ_oop_linear,u₀,trange,p2)
sol_oop = solve(prob_oop,EulerHeun(),dt=dt1,adaptive=false,save_noise=true)

@show length(sol_oop)

checkpoints=sol_oop.t

if length(unique(round.(checkpoints, digits=13))) != length(checkpoints)
  @warn "The given checkpoints are not unique. To avoid issues in the interpolation the checkpoints were redefined. You may want to check sol.t if default checkpoints were used."
  checkpoints = unique(round.(checkpoints, digits=13))
end
@show length(checkpoints)

intervals = map(tuple, @view(checkpoints[1:end-1]), @view(checkpoints[2:end]))
interval_end = intervals[end][end]
prob_oop.tspan[1] > interval_end && push!(intervals, (interval_end, prob_oop.tspan[1]))
cursor = lastindex(intervals)
interval = intervals[cursor]


_sol = deepcopy(sol_oop)
idx1 = searchsortedfirst(_sol.t, interval[1]-1000eps(interval[1]))
idx2 = searchsortedfirst(_sol.t, interval[2])
dt = abs(_sol.W.dt)
if dt < 1000eps(_sol.t[end])
  dt = interval[2] - interval[1]
end
@show dt
```